## Pan5294
CP205 Whole Capture BED file (build37) for coverage and variant calling

The bedfile consists of 

- VCP1 and VCP2 genes- to be padded +/-30bp for variant calling (+/-10 for coverage)
- VCP3/CP205 genes + non R code genes- to be padded +/- 10bp for variant calling and coverage
- SNPs (for FH PRS and know pathogenic SNPs in other regions) Provided as a list of rsIDs (CP205_pathogenicsnps_wrsids.txt)
- Genomic regions (some SNPs, some known clinically important non-coding regions) provided as a list of regions (CP205_pathogenic_snps_norsids.txt padded by +/-5bp)
- FXR1 non-coding region will be padded by +/-30bp, provided in FXR1_noncoding_region.txt. This region is for the muscle specific transcript described in https://doi.org/10.1038/s41467-019-08548-9

5'UTR included for VCP1 and VCP2 genes only; no padding

Lists produced by R Haines and N Pinto using information from Heidi and Michael. 

# Make VCP1 and VCP2 genes bedfile
Save Pan5294_VCP1_VCP2_transcripts.txt

# Save script used to generate bedfiles
Save refgene.py, refgene_readme.md, check_transcript.py, config.yaml and ncbiRefSeq.txt 

# Run refgene.py
Run refgene.py to create bedfiles for VCP1 and VCP2 genes with +/-30bp for variant calling and +/-10 for coverage.

# Make VCP3/CP205 bedfile + non R code genes
Save Pan5294_VCP3_CP205_transcripts.txt and Pan5294_nonRcode_genes.txt

Create bedfile for Pan5294_VCP3_CP205_transcripts.txt using refgene.py and save Pan5294_VCP3_CP205_query.json

example of refgene cmd:
python3 refgene.py --refgene ncbiRefSeq.txt --transcript-file /usr/local/src/mokaguys/development_area/mokabed/Bedmaker/Pan5294/Pan5294_VCP3_CP205_transcripts.txt --bed-format data --out Pan5294_VCP3_CP205_data.bed

The refseq gene was unable to find the following transcripts, because they are not MANE: Transcript(s) not found in ncbiRefSeq.txt: NM_000445.3, NM_001083899.1, NM_001130103.1, NM_001146040.1, NM_001390.4, NM_002584.2, NM_007171.3, NM_013334.3, NM_022068.2, NM_138569.2, NM_201380.2

A seperate bedfile will be created using these transcripts (VCP3_CP205_nonMANETranscripts.txt) and Pan5294_nonRcode_genes.txt

Remove the list of non-MANE transcripts from Pan5294_VCP3_CP205_transcripts.txt

Rerun refegene.py with modified Pan5294_VCP3_CP205_transcripts.txt

# Run check_transcripts.py

check_transcripts.py checks if all requested transcripts are present in the end bedfile generated by refgene.py

python3 /usr/local/src/mokaguys/development_area/mokabed/LiveBedfiles/TestArea_for_bed_generation_script --transcripts Pan5294_VCP1_VCP2_transcripts.txt --bed Pan5294_VCP1_VCP2_sambamba.bed
Result:All transcript lines are present in bed file.

python3 /usr/local/src/mokaguys/development_area/mokabed/LiveBedfiles/TestArea_for_bed_generation_script --transcripts Pan5294_VCP1_VCP2_transcripts.txt --bed Pan5294_VCP1_VCP2_data.bed
Result:All transcript lines are present in bed file.

python3 /usr/local/src/mokaguys/development_area/mokabed/LiveBedfiles/TestArea_for_bed_generation_script --transcripts Pan5294_VCP3_CP205_transcripts.txt --bed Pan5294_VCP3_CP205_data.bed
Result:All transcript lines are present in bed file.

python3 /usr/local/src/mokaguys/development_area/mokabed/LiveBedfiles/TestArea_for_bed_generation_script --transcripts Pan5294_VCP3_CP205_transcripts.txt --bed Pan5294_VCP3_CP205_sambamba.bed
Result:All transcript lines are present in bed file.

# Create bedfile for VCP3_CP205_nonMANETranscripts.txt and Pan5294_nonRcode_genes.txt

Run bedmaker and save the outputs

# Make additional regions bedfile

Save CP205_pathogenicsnps_wrsids.txt, CP205_pathogenic_snps_norsids.txt and FXR1_noncoding_region.txt

Run bedmaker and save Pan5294_additionalregions_query.json

# Remove chr prefix
bedfiles made by refgene.py had a 'chr' prefix and bedmaker bedfiles didn't. 'Chr' prefix removed using VScode

# Check for duplicated regions
A few genes have multiple transcripts requested, this has resulted in duplicated regions in the bedfiles.

Bedtools merge will be used to remove overlapping regions.

First combine Pan5294_VCP3_CP205_data.bed and VCP3_CP205_nonR_nonMANE_data.bed
NOTE: Pan5294_VCP1_VCP2_data.bed wasn't combined as we didn't want to merge the 5UTR regions.

cat VCP3_CP205_nonR_nonMANE_data.bed >> Pan5294_VCP3_CP205_data.bed

sort the bedfile
sort Pan5294_VCP3_CP205_data.bed -k1,1V -k2,2n -k3,3n > Pan5294_VCP3_CP205_data_sorted.bed

bedtools merge
-c- keep columns 4 and 5
-o first- choose the first value
bedtools merge -i Pan5294_VCP3_CP205_data_sorted.bed -c 4,5 -o first,first > Pan5294_VCP3_CP205_data_sorted_merged.bed

# Combine VCP1/VCP2 bedfile and additional regions bedfiles
cat Pan5294_VCP1_VCP2_data.bed Pan5294_additionalregions_data.bed >> Pan5294_VCP3_CP205_data_sorted_merged.bed

rename Pan5294_VCP3_CP205_data_sorted_merged.bed
mv Pan5294_VCP3_CP205_data_sorted_merged.bed Pan5294_data.bed

# sort bedfile
sort Pan5294_data.bed -k1,1V -k2,2n -k3,3n > Pan5294_data_sorted.bed; mv Pan5294_data.bed Pan5294_unsorted_data.bed; mv Pan5294_data_sorted.bed Pan5294_data.bed; rm Pan5294_unsorted_data.bed

# Check for duplicated regions in sambamba bedfiles

combine Pan5294_VCP3_CP205_sambamba.bed and VCP3_CP205_nonR_nonMANE_sambamba.bed
cat VCP3_CP205_nonR_nonMANE_sambamba.bed >> Pan5294_VCP3_CP205_sambamba.bed

sort the bedfile
sort -k1,1 -k2,2n -k3,3n Pan5294_VCP3_CP205_sambamba.bed > Pan5294_VCP3_CP205_sambamba_sorted.bed

combine regions
-c: keep columns 4-8
-o: keep first value for each column
bedtools merge -i Pan5294_VCP3_CP205_sambamba_sorted.bed -c 4,5,6,7,8 -o first,first,first,first,first >> Pan5294_VCP3_CP205_sambamba_sorted_merged.bed

# Combine VCP1/VCP2 bedfile and additional regions bedfiles
cat Pan5294_additionalregions_sambamba.bed Pan5294_VCP1_VCP2_sambamba.bed >> Pan5294_VCP3_CP205_sambamba_sorted_merged.bed

rename Pan5294_VCP3_CP205_data_sorted_merged.bed
mv Pan5294_VCP3_CP205_sambamba_sorted_merged.bed Pan5294_sambamba.bed

# sort sambamba bedfile
sort Pan5294_sambamba.bed -k1,1V -k2,2n -k3,3n > Pan5294_sambamba_sorted.bed; mv Pan5294_sambamba.bed Pan5294_unsorted_sambamba.bed; mv Pan5294_sambamba_sorted.bed Pan5294_sambamba.bed; rm Pan5294_unsorted_sambamba.bed

# Testing 
During testing Coverage Analysis using Sambamba v0.6.9 failed, '1-25870096-25870189' was on the same line as 'X-33192447-33192457' instead of a new line.

The error was fixed in VScode

# Re-sort sambamba.bed
sort Pan5294_sambamba.bed -k1,1V -k2,2n -k3,3n > Pan5294_sambamba_sorted.bed; mv Pan5294_sambamba.bed Pan5294_unsorted_sambamba.bed; mv Pan5294_sambamba_sorted.bed Pan5294_sambamba.bed; rm Pan5294_unsorted_sambamba.bed

# Testing 
Pan5294_data.bed was tested in DNAnexus using Moka Picard v1.2. The app completed successfully without any errors.

Pan5294_sambamba.bed was tested using sambamba_coverage_v2.0.3 The app completed successfully without any errors.

